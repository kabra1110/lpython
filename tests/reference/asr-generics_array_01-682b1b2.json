{
    "basename": "asr-generics_array_01-682b1b2",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/generics_array_01.py",
    "infile_hash": "c6df2de74d7c7d6c34034bba81ec72f26fb3fbab9f6651f0caced593",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-generics_array_01-682b1b2.stdout",
    "stdout_hash": "42b21ffdf7b39ef4d9ba925d85158d6f58e65b1395be155f2785cb6c",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}