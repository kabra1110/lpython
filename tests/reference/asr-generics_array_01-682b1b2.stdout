(TranslationUnit
    (SymbolTable
        1
        {
            _global_symbols:
                (Module
                    (SymbolTable
                        204
                        {
                            T:
                                (Variable
                                    204
                                    T
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (TypeParameter
                                        T
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            __asr_generic_f_0:
                                (Function
                                    (SymbolTable
                                        201
                                        {
                                            _lpython_return_variable:
                                                (Variable
                                                    201
                                                    _lpython_return_variable
                                                    []
                                                    ReturnVar
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            i:
                                                (Variable
                                                    201
                                                    i
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            lst:
                                                (Variable
                                                    201
                                                    lst
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Integer 4)
                                                        [(()
                                                        ())]
                                                        DescriptorArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    __asr_generic_f_0
                                    (FunctionType
                                        [(Array
                                            (Integer 4)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                        (Integer 4)]
                                        (Integer 4)
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 201 lst)
                                    (Var 201 i)]
                                    [(=
                                        (ArrayItem
                                            (Var 201 lst)
                                            [(()
                                            (IntegerConstant 0 (Integer 4))
                                            ())]
                                            (Integer 4)
                                            RowMajor
                                            ()
                                        )
                                        (Var 201 i)
                                        ()
                                    )
                                    (=
                                        (Var 201 _lpython_return_variable)
                                        (ArrayItem
                                            (Var 201 lst)
                                            [(()
                                            (IntegerConstant 0 (Integer 4))
                                            ())]
                                            (Integer 4)
                                            RowMajor
                                            ()
                                        )
                                        ()
                                    )
                                    (Return)]
                                    (Var 201 _lpython_return_variable)
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            _lpython_main_program:
                                (Function
                                    (SymbolTable
                                        203
                                        {
                                            
                                        })
                                    _lpython_main_program
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    [use_array]
                                    []
                                    [(SubroutineCall
                                        204 use_array
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            f:
                                (Function
                                    (SymbolTable
                                        199
                                        {
                                            _lpython_return_variable:
                                                (Variable
                                                    199
                                                    _lpython_return_variable
                                                    []
                                                    ReturnVar
                                                    ()
                                                    ()
                                                    Default
                                                    (TypeParameter
                                                        T
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            i:
                                                (Variable
                                                    199
                                                    i
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (TypeParameter
                                                        T
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            lst:
                                                (Variable
                                                    199
                                                    lst
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (TypeParameter
                                                            T
                                                        )
                                                        [(()
                                                        ())]
                                                        DescriptorArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    f
                                    (FunctionType
                                        [(Array
                                            (TypeParameter
                                                T
                                            )
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                        (TypeParameter
                                            T
                                        )]
                                        (TypeParameter
                                            T
                                        )
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        [(TypeParameter
                                            T
                                        )]
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 199 lst)
                                    (Var 199 i)]
                                    [(=
                                        (ArrayItem
                                            (Var 199 lst)
                                            [(()
                                            (IntegerConstant 0 (Integer 4))
                                            ())]
                                            (TypeParameter
                                                T
                                            )
                                            RowMajor
                                            ()
                                        )
                                        (Var 199 i)
                                        ()
                                    )
                                    (=
                                        (Var 199 _lpython_return_variable)
                                        (ArrayItem
                                            (Var 199 lst)
                                            [(()
                                            (IntegerConstant 0 (Integer 4))
                                            ())]
                                            (TypeParameter
                                                T
                                            )
                                            RowMajor
                                            ()
                                        )
                                        ()
                                    )
                                    (Return)]
                                    (Var 199 _lpython_return_variable)
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            use_array:
                                (Function
                                    (SymbolTable
                                        200
                                        {
                                            array:
                                                (Variable
                                                    200
                                                    array
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Integer 4)
                                                        [((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 1 (Integer 4)))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            x:
                                                (Variable
                                                    200
                                                    x
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    use_array
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    [__asr_generic_f_0]
                                    []
                                    [(=
                                        (Var 200 x)
                                        (IntegerConstant 69 (Integer 4))
                                        ()
                                    )
                                    (Print
                                        ()
                                        [(FunctionCall
                                            204 __asr_generic_f_0
                                            ()
                                            [((ArrayPhysicalCast
                                                (Var 200 array)
                                                FixedSizeArray
                                                DescriptorArray
                                                (Array
                                                    (Integer 4)
                                                    [((IntegerConstant 0 (Integer 4))
                                                    (IntegerConstant 1 (Integer 4)))]
                                                    DescriptorArray
                                                )
                                                ()
                                            ))
                                            ((Var 200 x))]
                                            (Integer 4)
                                            ()
                                            ()
                                        )]
                                        ()
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    _global_symbols
                    []
                    .false.
                    .false.
                ),
            lpython_builtin:
                (IntrinsicModule lpython_builtin),
            main_program:
                (Program
                    (SymbolTable
                        202
                        {
                            _lpython_main_program:
                                (ExternalSymbol
                                    202
                                    _lpython_main_program
                                    204 _lpython_main_program
                                    _global_symbols
                                    []
                                    _lpython_main_program
                                    Public
                                )
                        })
                    main_program
                    [_global_symbols]
                    [(SubroutineCall
                        202 _lpython_main_program
                        ()
                        []
                        ()
                    )]
                ),
            numpy:
                (Module numpy)
        })
    []
)
